Using Subset
Subset Size:  54962
Val Subset Size:  21789
Using CUDA
Epoch 1/50, Average Loss: 0.6691146183896948, Accuracy: 0.5887509571907819, Precision: 0.6012967180054373, Recall: 0.5205221741043795, F1 Score: 0.5574808141651213
Traceback (most recent call last):
  File "/nfs/home/students/t.reim/bachelor/pytorchtest/main.py", line 148, in <module>
    val_loss += criterion(val_outputs, val_labels.to(1))
RuntimeError: CUDA error: invalid device ordinal
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.